### DISCUSSION TITLE
Rendering quality

### DISCUSSION THREAD

**Discussion by pstrg** [sumatrapdfreader/sumatrapdf#3317] [General]
I do like SumatraPDF a lot and has long been the default app for PDF files.
Just by chance, noted lately that rendering of image-based documents is somewhat lacking if compared with other viewers.
To illustrate: please look at the screen capture in which a photocopied book is shown side by side with Sumatra on left, Okular on right.
Why does that happen?
Are there plans to integrate another rendering library, perhaps?

![sumatra-okular](https://user-images.githubusercontent.com/13485213/222261759-02e20614-2c01-4f73-9b1b-53e086228634.jpg)

**Comment by GJoe2** [comment]
I think they default it because it renders more faster, but still I dont know if there is a setting to improve the quality at the cost of render time.

**Comment by GitHubRulesOK** [comment]
Not related as such but there is a bug in FireFox/Mozilla PDF.js that chrome/Edge will embarrassingly destroy its PDF image rendering because of incompatibility's how Hardware acceleration is used on some browser devices (especially Dell+Intel) and not others! so much of rendering is down to hardware drivers and their wide range of settings.

SumatraPDF AFAIK slightly alters MuPDF default for AntiAlising which could result in slightly "fuzzier renders" in some cases, which for some character renders may or may not help, but as MuPDF is the core rendering engine any core display issues may need to be addressed there

Looking at the range of good and bad sources for that document I found some were better than others, that were less mutated, but I did not see any copy like yours so to make any difference a sample would be require for testing.
